src/client.ts

```diff
@@ -3534,12 +3534,19 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
     /**
      * @param {string} roomId
      * @param {string} topic
+     * @param {string} htmlTopic Optional.
      * @param {module:client.callback} callback Optional.
      * @return {Promise} Resolves: TODO
      * @return {module:http-api.MatrixError} Rejects: with an error response.
      */
-    public setRoomTopic(roomId: string, topic: string, callback?: Callback): Promise<ISendEventResponse> {
-        return this.sendStateEvent(roomId, EventType.RoomTopic, { topic: topic }, undefined, callback);
+    public setRoomTopic(
+        roomId: string,
+        topic: string,
+        htmlTopic?: string,
+        callback?: Callback,
+    ): Promise<ISendEventResponse> {
+        const content = ContentHelpers.makeTopicContent(topic, htmlTopic);
+        return this.sendStateEvent(roomId, EventType.RoomTopic, content, undefined, callback);
     }
```

Comment:
This is a breaking change

Reply:
Is it actually? The content is just added to. The `"topic": "..."` field is still there.

Or is adding fields already considered a breaking change?

Reply:
The `callback` parameter is no longer the 3rd parameter

Reply:
Oh, you're talking about the method, not the content. Sorry.

I guess still better to keep it like this because otherwise the two topic parameters would be spread out in the signature. But happy to swap the 3rd and 4th parameter if you think that'd be better.

Reply:
I have no real opinions, I commented to explain the addition of the X-Breaking-Change label

Reply:
Ideally we don't break the API surface - good spot @t3chguy 

@Johennes if we can make the third parameter support dual types like we do elsewhere, that'd be best. We'd be overriding the function definition to show the various signatures someone can use and the function itself would have to pick the types apart manually.

Reply:
Good idea. I added the overloads so this should not be a breaking change anymore.